flight/targets/pipxtreme/fw/pios_board.c

```diff
@@ -128,32 +164,155 @@ void PIOS_Board_Init(void) {
 	}
 #endif
 
+	// Configure the main port
+	HwTauLinkData hwTauLink;
+	HwTauLinkGet(&hwTauLink);
+
 	/*Initialize the USB device */
 	uintptr_t pios_usb_id;
 	PIOS_USB_Init(&pios_usb_id, &pios_usb_main_cfg);
 
-	PIOS_HAL_ConfigureHID(HWSHARED_USB_HIDPORT_USBTELEMETRY,
-			pios_usb_id, &pios_usb_hid_cfg);
+	PIOS_HAL_ConfigureHID(HWSHARED_USB_HIDPORT_USBTELEMETRY, pios_usb_id, &pios_usb_hid_cfg);
 
 	/* Configure the USB virtual com port (VCP) */
 #if defined(PIOS_INCLUDE_USB_CDC)
 	if (usb_cdc_present)
 	{
-		PIOS_HAL_ConfigureCDC(hwTauLink.VCPPort, pios_usb_id,
-				&pios_usb_cdc_cfg);
+		PIOS_HAL_ConfigureCDC(hwTauLink.VCPPort, pios_usb_id, &pios_usb_cdc_cfg);
 	}
 #endif
 
 	PIOS_HAL_ConfigurePort(hwTauLink.MainPort,
-			&pios_usart_serial_cfg, &pios_usart_com_driver,
-			/* no I2C, DSM, HSUM, SBUS, etc. */
-			NULL, NULL, NULL, PIOS_LED_ALARM,
-			NULL, NULL, 0, NULL, NULL, false);
-
-	// Configure the flexi port
-	switch (hwTauLink.PPMPort) {
-		case HWTAULINK_PPMPORT_PPM:
-			{
+	    &pios_usart_serial_cfg, &pios_usart_com_driver,
+	    /* no I2C, DSM, HSUM, SBUS, etc. */
+	    NULL, NULL, NULL, PIOS_LED_ALARM,
+	    NULL, NULL, 0, NULL, NULL, false);
+
+	// Update the com baud rate.
+	uint32_t comBaud = 9600;
+	switch (hwTauLink.ComSpeed) {
+	case HWTAULINK_COMSPEED_4800:
+		comBaud = 4800;
+		break;
+	case HWTAULINK_COMSPEED_9600:
+		comBaud = 9600;
+		break;
+	case HWTAULINK_COMSPEED_19200:
+		comBaud = 19200;
+		break;
+	case HWTAULINK_COMSPEED_38400:
+		comBaud = 38400;
+		break;
+	case HWTAULINK_COMSPEED_57600:
+		comBaud = 57600;
+		break;
+	case HWTAULINK_COMSPEED_115200:
+		comBaud = 115200;
+		break;
+	}
+
+	const struct pios_rfm22b_cfg *rfm22b_cfg = PIOS_BOARD_HW_DEFS_GetRfm22Cfg(bdinfo->board_rev);
+	PIOS_HAL_ConfigureRFM22B(hwTauLink.Radio, bdinfo->board_type,
+	    bdinfo->board_rev, hwTauLink.MaxRfPower,
+	    hwTauLink.MaxRfSpeed, NULL, rfm22b_cfg,
+	    hwTauLink.MinChannel, hwTauLink.MaxChannel,
+	    hwTauLink.CoordID, 0);
+
+	if (bdinfo->board_rev == TAULINK_VERSION_MODULE) {
+		// Configure the main serial port function
+		switch (hwTauLink.BTPort) {
+		case HWTAULINK_BTPORT_TELEMETRY:
+		{
+			// Note: if the main port is also on telemetry the bluetooth
```

Comment:
when I used combridge it was normally because i wanted to send commands directly to the bt module and configure it that way so I wanted it at the previous (or default) configuration